presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -291,6 +415,98 @@ public static double round(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(S
         return Math.floor(num * factor + 0.5) / factor;
     }
 
+    private static SqlScalarFunction decimalRoundNFunction()
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name("round")
+                .literalParameters("num_precision", "num_scale")
+                .argumentTypes("decimal(num_precision, num_scale)", StandardTypes.BIGINT)
+                .returnType("decimal(num_precision, num_scale)")
+                .build();
+        return SqlScalarFunction.builder(MathFunctions.class)
+                .signature(signature)
+                .description("remainder of given quotient")
+                .methods("roundShortDecimal", "roundLongDecimal")
+                .extraParameters(MathFunctions::decimalRoundNExtraParameters)
+                .build();
+    }
+
+    private static SqlScalarFunction decimalRoundFunction()
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name("round")
+                .literalParameters("num_precision", "num_scale")
+                .argumentTypes("decimal(num_precision, num_scale)")
+                .returnType("decimal(num_precision, num_scale)")
```

Comment:
if we choose to split the implementations of round(x) and round(x, n) then we can do this, they will have different behaviours then (first one adjusting the precision, the latter leaving as is) but as we agreed we want to fix the latter eventually so we'll follow this path.